ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala

```
@@ -324,7 +341,9 @@ object BrokerCollie {
     private[this] var jmxPort: Int = CommonUtils.availablePort()
 
     override protected def doCopy(clusterInfo: BrokerClusterInfo): Unit = {
-      zookeeperClusterName(clusterInfo.zookeeperClusterName)
+      zookeeperClusterName(
```

How about coping it only if the input bk cluster has zk cluster?

---

You mean `doCopyOrDoNothing()` ?

---

yep

---

I think we can "fail fast" in here ; Since `doNothing` will cause exception in actual action (in here, it's addNode). 
Since this method is actual "do copy from actual cluster", Maybe absent zk cluster case not exists?

---

You are right :)